fog/ingest/server/src/controller.rs

```diff
@@ -222,8 +259,7 @@ where
     pub fn new_keys(&self) -> Result<IngestSummary, Error> {
         let mut state = self.get_state();
         self.new_keys_inner(&mut state)?;
-        self.write_state_file_inner(&mut state);
-        Ok(self.get_ingest_summary())
+        Ok(self.get_ingest_summary_inner(&mut state))
```

Comment:
this was likely a bug -- this function was calling `get_ingest_summary` while holding the state lock mutex, so i would expect a deadlock

Reply:
there needs to be a test that exercises this API, clearly there wasn't one before this

Reply:
done